mqtt-v5/src/topic.rs

```diff
@@ -17,6 +17,13 @@ pub enum TopicFilter {
 
 /// A topic name publishers use when sending MQTT messages.
 /// Cannot contain wildcards.
+///
+/// # Example
+/// ```rust
+/// use std::str::FromStr;
+/// use mqtt_v5::topic::Topic;
+/// let topic = Topic::from_str("my_topic").unwrap();
+/// ```
```

Comment:
I appreciate the documentation, but let's leave this for a separate PR and have this one just focus on fixing the overflow error.